File: src/main/java/com/thinkfirst/repository/LearningPathRepository.java
package com.thinkfirst.repository;

import com.thinkfirst.model.LearningPath;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface LearningPathRepository extends JpaRepository<LearningPath, Long> {
    
    Optional<LearningPath> findByQuizIdAndChildId(Long quizId, Long childId);
    
    Optional<LearningPath> findByIdAndChildId(Long id, Long childId);
}

File: android/app/src/main/java/com/thinkfirst/android/data/repository/LearningPathRepository.kt
package com.thinkfirst.android.data.repository

import com.thinkfirst.android.data.api.ThinkFirstApi
import com.thinkfirst.android.data.model.LearningPath
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class LearningPathRepository @Inject constructor(
    private val api: ThinkFirstApi
) {
    suspend fun completeLesson(lessonId: Long, childId: Long): LearningPath {
        return api.completeLesson(lessonId, childId)
    }

    suspend fun getLearningPath(learningPathId: Long, childId: Long): LearningPath {
        return api.getLearningPath(learningPathId, childId)
    }
}

